qrcode/util.py

```diff
@@ -189,51 +189,40 @@ def _lost_point_level1(modules, modules_count):
     lost_point = 0
 
     modules_range = xrange(modules_count)
-    row_range_first = (0, 1)
-    row_range_last = (-1, 0)
-    row_range_standard = (-1, 0, 1)
-
-    col_range_first = ((0, 1), (1,))
-    col_range_last = ((-1, 0), (-1,))
-    col_range_standard = ((-1, 0, 1), (-1, 1))
+    container = [0] * (modules_count + 1)
 
     for row in modules_range:
-
-        if row == 0:
-            row_range = row_range_first
-        elif row == modules_count-1:
-            row_range = row_range_last
-        else:
-            row_range = row_range_standard
-
+        this_row = modules[row]
+        previous_color = this_row[0]
+        length = 0
         for col in modules_range:
-
-            sameCount = 0
-            dark = modules[row][col]
-
-            if col == 0:
-                col_range = col_range_first
-            elif col == modules_count-1:
-                col_range = col_range_last
+            if this_row[col] == previous_color:
+                length += 1
             else:
-                col_range = col_range_standard
-
-            for r in row_range:
-
-                row_offset = row + r
-
-                if r != 0:
-                    col_idx = 0
-                else:
-                    col_idx = 1
-
-                for c in col_range[col_idx]:
-
-                    if dark == modules[row_offset][col + c]:
-                        sameCount += 1
+                if length >= 5:
+                    container[length] += 1
+                length = 1
+                previous_color = this_row[col]
+        if length >= 5:
+            container[length] += 1
+
+    for col in modules_range:
+        #this_col = [this_row[col] for this_row in modules] Slow, commented out.
```

Comment:
If so, I'll drop it out.
Moving `this_row` out of the inner for loop can increase the speed a bit. But `this_col` will be slower. I did that to leave a message so others won't move `this_col` out again for optimization.